### PR TITLE
Multiutil.cpp cleanup#1

### PR DESCRIPTION
**Pull request by JohnAFernandez**
Catch-all cleanup of multiutil.cpp. I did my best on these changes, but I am not attached to them, so please make any necessary requests for further changes.  No functionality at all should be changed.

I tried to split the commits into three categories: 

Category 1) the removal of 1 unused function with functionality that exists elsewhere, code that has been commented out since the source code was released, and 2 globals that were there to be stored by the engine and incremented and nothing else. (Their discovery during bug hunting -- and my general mental exhaustion from the same -- inspired this PR)

Category 2) changing NULL to nullptr

Category 3) Changing int3()'s into assertion()'s, error()'s, or verify()'s

### CODE REVIEW

File: code/network/multiutil.cpp [hunk 1]
```diff
@@ -842,7 +843,7 @@ int multi_create_player( int net_player_num, player *pl, const char* name, net_a
 void multi_make_player_ai( object *pobj )
 {
 
-	Assert ( pobj != NULL );
+	Assert(pobj != NULL);
```
Comment (CasparKielwein) [inline code/network/multiutil.cpp]: you missed a nullptr here

File: code/network/multiutil.cpp [hunk 2]
```diff
@@ -2945,14 +2928,13 @@ void multi_get_mission_checksum(const char *filename)
 
 char multi_unit_to_char(float unit)
 {
+	Verify((unit > 1.0f) && (unit < -1.0f), "Newly implemented function multi_unit_to_char() was passed a value %f, that it cannot handle.", unit);
```
Comment (CasparKielwein) [inline code/network/multiutil.cpp]: here and below:
the Verify macro only supports a single argument.
You could use the old hack/trick of "anding" the check and the message.
The format is not possibly unfortunately. Asserts and their variants like Verify usually only support string literals.

```Verify((unit > 1.0f) && (unit < -1.0f) && "Newly implemented function multi_unit_to_char() was passed a value, that it cannot handle.");```

Comment (JohnAFernandez) [reply code/network/multiutil.cpp]: Ah. I probably should avoid using a hack in a cleanup.  I'll just switch to Assert.  Whoever wants to implement this function can do it using debug builds.

File: code/network/multiutil.cpp (hunk 1 above)
Comment (JohnAFernandez) [reply code/network/multiutil.cpp]: I think I have this replaced with an assertion but I forgot to commit the changes.

**Comment by JohnAFernandez** [comment]
Just as a note, it looks like in a few places I'm just removing int3()'s, but there was no way to reach an error value in the first place for those.  A lot of this file seems to have been a WIP when the game was released, so they were probably planning on adding some.  Most of the problems for the observer and player create functions would come from bad arguments being passed. There's also no packet being sent or received, so Assertions work really well here.

**Comment by JohnAFernandez** [comment]
One multi file down, 73 to go. >.>

**Pull request merged**